src/test/java/org/eolang/speco/MainTest.java

```diff
@@ -57,44 +59,53 @@ public final class MainTest {
      */
     private final Path eos = this.tests.resolve("eo");
 
-    @Test
-    public void convertsFromXmir(@TempDir final Path temp) throws IOException {
-        final Path output = this.runSpeco(false, temp);
-        for (final Path path : Files.newDirectoryStream(output)) {
-            final List<String> expected = Files.readAllLines(path);
-            final List<String> produced = Files.readAllLines(temp.resolve(path.getFileName()));
-            Assertions.assertEquals(expected, produced);
-        }
+    @ParameterizedTest
+    @ValueSource(strings = {"simple"})
+    public void convertsFromXmir(final String name, @TempDir final Path temp) throws IOException {
+        final Path output = this.runSpeco(this.xmirs.resolve(name), temp, false);
+        this.compare(temp, output);
     }
 
-    @Test
-    public void convertsFromEo(@TempDir final Path temp) throws IOException {
-        final Path output = this.runSpeco(true, temp);
-        for (final Path path : Files.newDirectoryStream(output)) {
+    @ParameterizedTest
+    @ValueSource(strings = {"booms", "pets"})
+    public void convertsFromEo(final String name, @TempDir final Path temp) throws IOException {
+        final Path base = this.eos.resolve(name);
+        final Path output = this.runSpeco(base, temp, true);
+        this.compare(temp, output);
+        final List<String> expected = Files.readAllLines(base.resolve("result.txt"));
+        final List<String> produced = this.exec(temp.toString());
+        Assertions.assertEquals(expected, produced);
+    }
+
+    /**
+     * Compares two directpries.
+     *
+     * @param first Directory
+     * @param second Directory
+     * @throws IOException Iff IO error
+     */
+    private void compare(final Path first, final Path second) throws IOException {
+        for (final Path path : Files.newDirectoryStream(second)) {
             final List<String> expected = Files.readAllLines(path);
-            final List<String> produced = this.exec(temp.toString());
+            final Path out = first.resolve(path.getFileName());
+            final List<String> produced = Files.readAllLines(out);
             Assertions.assertEquals(expected, produced);
         }
     }
 
     /**
      * Runs Speco.
      *
-     * @param eolang Iff input is EO
+     * @param base Path to the base input dir
      * @param temp Path to the temporary output dir
+     * @param iseo Iff input is EO
      * @return Path to the reference output dir
      * @throws IOException Iff IO error
      */
-    private Path runSpeco(final boolean eolang, final Path temp) throws IOException {
-        final Path base;
-        if (eolang) {
-            base = this.eos;
-        } else {
-            base = this.xmirs;
-        }
+    private Path runSpeco(final Path base, final Path temp, final boolean iseo) throws IOException {
```

Comment:
@advasileva The method could be `static`.
Probably we can also inline some variables here.